src/test/java/dk/cngroup/lentils/service/StatusServiceIntegrationTest.java

```diff
@@ -41,4 +46,66 @@ public void initializeStatusForTeamAndCypher() {
 
         assertEquals(CypherStatus.LOCKED, statusService.getCypherStatusByTeamAndCypher(team, cypher));
     }
+
+    @Test
+    public void markCypherWhenNextCyphersAreLocked() {
+        Team team = objectGenerator.generateValidTeam();
+        teamRepository.saveAndFlush(team);
+        final Cypher cypher1 = getCypherWithStageNumber(101);
+        Status status1 = setStatusForCypherAndTeam(cypher1, team, CypherStatus.PENDING);
+        final Cypher cypher2 = getCypherWithStageNumber(102);
+        Status status2 = setStatusForCypherAndTeam(cypher2, team, CypherStatus.LOCKED);
+        final Cypher cypher3 = getCypherWithStageNumber(103);
+        Status status3 = setStatusForCypherAndTeam(cypher3, team, CypherStatus.LOCKED);
+
+        statusService.markCypher(cypher1, team, CypherStatus.SOLVED);
+
+        assertEquals(CypherStatus.PENDING, statusService.getCypherStatusByTeamAndCypher(team, cypher2));
+    }
+
+    @Test
+    public void markCypherWhenNextCyphersAreNotLocked() {
+        Team team = objectGenerator.generateValidTeam();
+        teamRepository.saveAndFlush(team);
+        final Cypher cypher1 = getCypherWithStageNumber(101);
+        Status status1 = setStatusForCypherAndTeam(cypher1, team, CypherStatus.SOLVED);
+        final Cypher cypher2 = getCypherWithStageNumber(102);
+        Status status2 = setStatusForCypherAndTeam(cypher2, team, CypherStatus.SOLVED);
+        final Cypher cypher3 = getCypherWithStageNumber(103);
+        Status status3 = setStatusForCypherAndTeam(cypher3, team, CypherStatus.PENDING);
+
+        statusService.markCypher(cypher1, team, CypherStatus.PENDING);
+
+        assertEquals(CypherStatus.SOLVED, statusService.getCypherStatusByTeamAndCypher(team, cypher2));
```

Comment:
Tady bych přidal assert, že ta druhá šifra opravdu zůstává ve stavu solved.
To je tady vlastně to, co potřebuješ zkontrolovat.

Reply:
myslim ze tam ten assert mam - pridala jsem tedy assert na kontrolu prvni sifry, ze se prepla do pending.